backend/src/Services/SigningService.cs

```diff
@@ -2,56 +2,84 @@
+/// <param name="logger">      Instance of <see cref="ILogger"/> </param>
 public class SigningService(
     AppSettings appSettings,
     ILogger<ISigningService> logger) : ISigningService
 {
-    //Export private subkey: gpg --output private.pgp --armor --export-secret-key C97074097892B88EF6D23BA9E5D03DA76EDC407D
     private string _fingerprint = "";
+    private const string FILENAME = "dataToSign";
 
-    public async Task<bool> ImportPrivateKey()
+    /// <inheritdoc/>
+    public async Task<bool> Initialize()
     {
-        //await ExecuteCommand();
-        var (success, outout) = await ExecuteGnuCommand($"gpg --batch --passphrase Freiburg2005 --import ../gpg-keys/{appSettings.GnupgPrivateKeyFilename}");
+        // Execute command to import private key
+        var (success, outout) = await ExecuteGnuCommand($"gpg --batch --passphrase {appSettings.GnupgPrivateKeyPassphrase} --import ../gpg-keys/{appSettings.GnupgPrivateKeyFilename}");
         if (success)
         {
-            _fingerprint = await GetFingerprint();
+            // Extract fingerprint
+            _fingerprint = await GetFingerprintFromKeyList();
             if (!String.IsNullOrEmpty(_fingerprint))
             {
-                logger.LogDebug("GNUPG Fingerprint: {Fingerprint}", _fingerprint);
-                return true;
+                logger.Fingerprint(_fingerprint);
+            }
+            else
+            {
+                logger.FingerprintError();
+                success = false;
             }
         }
-        return false;
+        return success;
     }
 
-    public async Task<bool> SignData(string data)
+    /// <inheritdoc/>
+    public string GetFingerprint()
     {
-        //await ExecuteCommand();
-        //await ExecuteGnuCommand($"echo $(tty)");
-        var filepath = WriteDataToFile(data);
-        await ExecuteGnuCommand($"echo $(tty)");
-        await ExecuteGnuCommand($"echo $GPG_TTY");
-        await ExecuteGnuCommand($"gpgconf --list-dirs");
-        // gpg --detach-sig --armor --local-user <SIGNING_KEY_FINGERPRINT> response.json
-        //var (success, outout) = await ExecuteGnuCommand($"gpg --output {Path.GetFileName(filepath)}.asc --detach-sig --armor {Path.GetFileName(filepath)}");
-        //var (success, outout) = await ExecuteGnuCommand($"gpg --batch --passphrase Freiburg2005 --detach-sig --armor --local-user {_fingerprint} {Path.GetFileName(filepath)}");
-        await ExecuteGnuCommand($"echo \"test\" | gpg --pinentry-mode loopback --batch --passphrase Freiburg2005 --clearsign");
-        var (success, outout) = await ExecuteGnuCommand($"gpg --pinentry-mode loopback --batch --passphrase Freiburg2005 --clearsign {Path.GetFileName(filepath)}");
-
-        var signature = ReadSignatureFromFile(filepath);
-        logger.LogDebug("GNUPG Signature: {Signature}", signature);
-        //RemoveFiles(filepath);
-        return success;
+        return _fingerprint;
```

Comment:
Why not have the property `Fingerprint` with a public getter and a private setter?